respy/tests/random_model.py

```diff
@@ -169,6 +158,16 @@ def drop_agents_obs(agent):
         data_subset = df
 
     # We can restrict the information to observed entities only.
-    data_subset = data_subset[DATA_LABELS_EST]
+    labels, _ = _generate_column_labels_estimation(options)
+    data_subset = data_subset[labels]
 
     return data_subset
+
+
+def _update_dict_w_dict(d, u):
```

Comment:
This function needs a docstring that explains how it differs from a normal `dict.update`. Also I would prefer to call it `_update_nested_dictionary` and to call the arguments `dict_` or `dictionary` and `other`.